src/compiler/checker.ts

```diff
@@ -29143,6 +29142,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return errorType;
         }
 
+        if (classDeclarationExtendsNull(classLikeDeclaration)) {
+            return isCallExpression ? errorType : nullWideningType;

```

Comment:
The error itself for calls is already raised by `checkConstructorDeclarationDiagnostics`. I thought about moving it here but I concluded that leaving it there is nice because it only errors on the very first super call in the constructor. In here, we'd be erroring on all calls.